### DISCUSSION TITLE
Home Assistant 2024.12

### DISCUSSION THREAD

**Discussion by paulraison** [community-scripts/ProxmoxVE#708] [Ideas]
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

As of Home Assistant 2024.12, it requires Python 3.13 to be installed. The current Home Assistant Core install script only installs Python 3.12 which is the default for Ubuntu 24.04.

### What settings are you currently utilizing?

- [X] Default Settings
- [ ] Advanced Settings

### Which Linux distribution are you employing?

Ubuntu 24.04

**Comment by michelroegl-brunner** [comment]
Wich script are you refering to specificly?

**Comment by paulraison** [comment]
It's the Home Assistant Core script.

bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/homeassistant-core.sh)"

**Comment by michelroegl-brunner** [comment]
ok, i will have a look :)

**Comment by havardthom** [comment]
I'm converting this to a discussion since this is not an issue atm, the Home Assistant scripts works fine as is.

**Comment by Claudio1L** [comment]
same issue here. 

I'm been able to install python 3.13 manually, but still having error in the log

 Support for the running Python version 3.12.3 is deprecated and will be removed in Home Assistant 2025.2; Please upgrade Python to 3.13 

causing ha "default_config" not loading correctly

**Reply by MickLesk** [reply]
Have you removed the old Python before? Then Install the new and reboot

**Reply by Claudio1L** [reply]
yes, just did it. still having 
![image](https://github.com/user-attachments/assets/8f82721e-0ea7-4657-a74b-d3dc7a016fcd)

can't understand why

**Reply by havardthom** [reply]
It is just a warning that it is deprecated, should not affect anything until Home Assistant 2025.2 is out

**Reply by elad-bar** [reply]
I will try to find a solution for that during weekend,
My direction is to use asdf to allow multiple python version running together while only one defined as global,
It will provide backward compatability in case needed as well as future support to any python version that will get released.

**Comment by elad-bar** [comment]
i also have the same issue, i would expect the update script to handle it, but it just upgrade the direct package and not the prerequisites,
can you please refer me to process of how to uninstall python 3.12 and install new one?
i will add it to update script and share it

thanks

**Reply by michelroegl-brunner** [reply]
There is already a PR, please have a look befor you submit anything. https://github.com/community-scripts/ProxmoxVE/pull/696

**Reply by elad-bar** [reply]
I was writing about update script not installing new container, if that's the same you can simply share a PR that is not available in this thread rather commanding me not to post anything...

**Reply by michelroegl-brunner** [reply]
I'm sorry if i sounded rude, this was not my intention. English is not my first language, so my spelling might be off.

I did not want to command you around, rather i wanted to refer you to the discussion in the PR, if we should wait a bit for things to settel or not befor updating anything, as the discussion were mainly about the dependencys and the availability at the moment. This includs not only the install script, but also the update as well.

It was never my intention to command anyone around!

**Reply by elad-bar** [reply]
That's ok, getting back to my question, will it work for an update script or just for new install?
(As I understood that's 2 different scripts)
Thanks

**Reply by towerhand** [reply]
Wait for a new install script to create a new container and use the copy config script to move things over.

Edit: This is what tteck recommended when python 3.11 was deprecated. https://github.com/tteck/Proxmox/discussions/2458

**Reply by michelroegl-brunner** [reply]
> That's ok, getting back to my question, will it work for an update script or just for new install?
> 
> (As I understood that's 2 different scripts)
> 
> Thanks

There will be an update for the update script as well, when all issues with dependencies and so are worked out.

**Comment by Pyromane** [comment]
I read in the release/change logs that the problem should be fixed: https://github.com/community-scripts/ProxmoxVE/releases/tag/2025-01-04

Therefore I used 
`bash -c “$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/homeassistant-core.sh)”`
I created a new LXC and downloaded the data using
`bash -c “$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/misc/copy-data/home-assistant-core-copy-data-home-assistant-core.sh)”`
migrated.

I started the container, deleted the logs and restarted it only to read the following error message again:
```
Logger: homeassistant.bootstrap
Source: bootstrap.py:505
First occurred: 15:12:03 (1 occurrences)
Last logged: 15:12:03

Support for the running Python version 3.12.3 is deprecated and will be removed in Home Assistant 2025.2; Please upgrade Python to 3.13
```

**Comment by SergioRius** [comment]
FYI, Installing as ubuntu 24.10 base, so it installs python 3.13, is broken ATM:
```
unable to create CT 5041 - unsupported Ubuntu version '24.10'
bash: line 201: exit: A problem occured while trying to create container.: numeric argument required
```
Trying to create a fresh CT with ubuntu 24.04 always installs python 3.12.3 and HA 2024.12 (and during the process shows several issues with Figlet/fonts and ends crashing about it).

Trying to update an existing 2024.12 instance does nothing. I haven't found a way to upgrade from there.

**Reply by MickLesk** [reply]
And Homeassistant sleeps. Its only available for 24.04 yet

**Reply by cavazos-apps** [reply]
Manually installing Home Assistant  Core on an Ubuntu 24.10 LXC works as expected and shows no Python warning and the correct version. How is 24.04 required?

**Reply by MickLesk** [reply]
In there Docs. They are lagging behind.

**Comment by MickLesk** [comment]
Ive build an develop merge.
Need tester for ubuntu 24.10 version (Default only)

```
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/homeassistant-core.sh)"
```


PS: _/dev/stdin: line 279: syntax error: unexpected end of file_ is not an script issue, its and develop thing
& **! DONT UPDATE YOUR LXC WITH THIS FILE / ONLY CREATE AN NEW ONE !**

**Reply by cavazos-apps** [reply]
It is still installing 2024 as uv is using Python 3.12 using `bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/develop/ct/homeassistant-core.sh)"`

```
Using Python 3.12.7 environment at: packages 
Resolved 96 packages in 1.42s
   Built pyric==0.1.6.3
   Built webrtcvad==2.0.10
   Built fnvhash==0.1.0
   Built mysqlclient==2.2.6
Prepared 95 packages in 4.18s
Installed 96 packages in 370ms
 + acme==3.0.1
 + aiodns==3.2.0
 + aiohappyeyeballs==2.4.4
 + aiohasupervisor==0.2.1
 + aiohttp==3.11.11
 + aiohttp-cors==0.7.0
 + aiohttp-fast-zlib==0.2.0
 + aiooui==0.1.7
 + aiosignal==1.3.2
 + aiozoneinfo==0.2.1
 + anyio==4.8.0
 + astral==2.2
 + async-interrupt==1.2.0
 + async-timeout==5.0.1
 + atomicwrites-homeassistant==1.4.1
 + attrs==24.2.0
 + awesomeversion==24.6.0
 + bcrypt==4.2.0
 + bleak==0.22.3
 + bleak-retry-connector==3.6.0
 + bluetooth-adapters==0.20.2
 + bluetooth-auto-recovery==1.4.2
 + bluetooth-data-tools==1.20.0
 + boto3==1.35.94
 + botocore==1.35.94
 ⠏==0.3.0et
 + certifi==2024.12.14
 + cffi==1.17.1
 + charset-normalizer==3.4.1
 + ciso8601==2.3.1
 + cryptography==43.0.1
 + dbus-fast==2.28.0
 + envs==1.4
 + fnv-hash-fast==1.0.2
 + fnvhash==0.1.0
 + frozenlist==1.5.0
 + greenlet==3.1.1
 + h11==0.14.0
 + habluetooth==3.7.0
 + hass-nabucasa==0.86.0
 + home-assistant-bluetooth==1.13.0
 + homeassistant==2024.12.5
 + httpcore==1.0.7
 + httpx==0.27.2
 + idna==3.10
 + ifaddr==0.2.0
 + isal==1.7.1
 + jinja2==3.1.4
 + jmespath==1.0.1
 + josepy==1.14.0
 + lru-dict==1.3.0
 + markupsafe==3.0.2
 + mashumaro==3.15
 + multidict==6.1.0
 + mysqlclient==2.2.6
 + orjson==3.10.12
 + packaging==24.2
 + pillow==11.0.0
 + propcache==0.2.1
 + psutil==6.1.1
 + psutil-home-assistant==0.0.1
 + psycopg2-binary==2.9.10
 + pycares==4.5.0
 + pycognito==2024.5.1
 + pycparser==2.22
 + pyjwt==2.10.1
 + pyopenssl==24.2.1
 + pyrfc3339==2.0.1
 + pyric==0.1.6.3
 + python-dateutil==2.9.0.post0
 + python-slugify==8.0.4
 + pytz==2024.2
 + pyyaml==6.0.2
 + requests==2.32.3
 + s3transfer==0.10.4
 + securetar==2024.11.0
 + setuptools==75.7.0
 + six==1.17.0
 + sniffio==1.3.1
 + snitun==0.39.1
 + sqlalchemy==2.0.36
 + text-unidecode==1.3
 + typing-extensions==4.12.2
 + tzdata==2024.2
 + uart-devices==0.1.0
 + ulid-transform==1.0.2
 + urllib3==1.26.20
 + usb-devices==0.4.5
 + uv==0.5.4
 + voluptuous==0.15.2
 + voluptuous-openapi==0.0.5
 + voluptuous-serialize==2.6.0
 + webrtc-models==0.3.0
 + webrtcvad==2.0.10
 + wheel==0.45.1
 + yarl==1.18.3
```

**Reply by MickLesk** [reply]
I don't care about the version of HA, it is delivered via the repo. You have to try Python in a few minutes, the push sometimes takes a while

**Reply by cavazos-apps** [reply]
```
Setup Python3
 ⠴error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.13/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
 ⠦
[ERROR] in line 65: exit code 0: while executing command $STD pip install uv
```

**Reply by MickLesk** [reply]
already fixed since 10min. i say wait a little bit

**Reply by cavazos-apps** [reply]
As of 7e93e5e, `[ERROR] in line 67: exit code 0: while executing command python3 venv . &> /dev/null`. Local testing indicates `python3 venv . &>/dev/null` is not creating the venv. `python3 -m venv . &>/dev/null` fixes this error.

**Reply by MickLesk** [reply]
done

**Reply by cavazos-apps** [reply]
Getting closer. Now mysqlclient fails to install via `python3 -m pip install webrtcvad wheel homeassistant==2025.1.1 mysqlclient psycopg2-binary isal`:

```
Collecting webrtcvad
  Using cached webrtcvad-2.0.10.tar.gz (66 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting wheel
  Using cached wheel-0.45.1-py3-none-any.whl.metadata (2.3 kB)
Collecting homeassistant
  Using cached homeassistant-2025.1.1-py3-none-any.whl.metadata (4.8 kB)
Collecting mysqlclient
  Using cached mysqlclient-2.2.6.tar.gz (91 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [35 lines of output]
      /bin/sh: 1: pkg-config: not found
      /bin/sh: 1: pkg-config: not found
      /bin/sh: 1: pkg-config: not found
      /bin/sh: 1: pkg-config: not found
      Trying pkg-config --exists mysqlclient
      Command 'pkg-config --exists mysqlclient' returned non-zero exit status 127.
      Trying pkg-config --exists mariadb
      Command 'pkg-config --exists mariadb' returned non-zero exit status 127.
      Trying pkg-config --exists libmariadb
      Command 'pkg-config --exists libmariadb' returned non-zero exit status 127.
      Trying pkg-config --exists perconaserverclient
      Command 'pkg-config --exists perconaserverclient' returned non-zero exit status 127.
      Traceback (most recent call last):
        File "/srv/homeassistant/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
          ~~~~^^
        File "/srv/homeassistant/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
        File "/srv/homeassistant/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-e_erplk6/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-e_erplk6/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
          ~~~~~~~~~~~~~~^^
        File "/tmp/pip-build-env-e_erplk6/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
          ~~~~^^^^^^^^^^^^^^^^
        File "<string>", line 155, in <module>
        File "<string>", line 49, in get_config_posix
        File "<string>", line 28, in find_package_name
      Exception: Can not find valid pkg-config name.
      Specify MYSQLCLIENT_CFLAGS and MYSQLCLIENT_LDFLAGS env vars manually
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

Additionally, `==2025.1.1` is not needing when pip is using Python 3.13. Logs show `python3 -m pip install homeassistant' installs the most recent version (2025.1.1). 🙂

**Reply by cavazos-apps** [reply]
Adding these dependencies fixes the error:

`apt-get install default-libmysqlclient-dev pkg-config`

https://pypi.org/project/mysqlclient/

**Reply by MickLesk** [reply]
default-libmysqlclient-dev is death, i try it with libmariadb-dev

**Reply by cavazos-apps** [reply]
Just want to say thank you for taking the time to run through all of this. Appreciate the quick changes!

**Reply by cavazos-apps** [reply]
<img width="608" alt="image" src="https://github.com/user-attachments/assets/8b7a98ec-720f-4dbd-9d53-c396aa922b24" />

Success! It looks like the port was changed from `8123` to `8132` in the message so the IP in the final message is incorrect. `8123` is the correct port.

**Reply by MickLesk** [reply]
its an issue of service file. I push the changes tomorrow, then it works with the correct port (uv remove pip3 and changed servie)

![image](https://github.com/user-attachments/assets/ccaa565d-9d40-4653-ae4b-fca84bf63121)

**Reply by cavazos-apps** [reply]
![image](https://github.com/user-attachments/assets/96bad611-2510-4ede-8240-be315f21fca3)

**Comment by dmshimself** [comment]
The testing version worked well for me and all my entities came good.  A couple of the HACS updates which had a minimum supported version that was after 2024.* came good too.  Great progress.

**Reply by WildeRNS** [reply]
did you updated or fresh install with data-copy?

**Reply by MickLesk** [reply]
i hope a fresh install ^^

**Reply by dmshimself** [reply]
Yes indeed a fresh core install and then the handy scripts to do the data copy.

**Reply by WildeRNS** [reply]
> i hope a fresh install ^^

Installed fresh and copy by script data - everything ok.
But now question: how to update? Wait for script update?

**Reply by MickLesk** [reply]
I think wait for Script. But i need a tester, because ive need to write an Migration or you all need the Copy Script with an new LXC. 22.04 to 24.04/24.10 is an big Change 

Ive only running an Homeassistant OS VM

**Reply by WildeRNS** [reply]
> I think wait for Script. But i need a tester, because ive need to write an Migration or you all need the Copy Script with an new LXC. 22.04 to 24.04/24.10 is an big Change
> 
> Ive only running an Homeassistant OS VM

I can test in future, when new version of HA will be available.

**Reply by towerhand** [reply]
> I think wait for Script. But i need a tester, because ive need to write an Migration or you all need the Copy Script with an new LXC. 22.04 to 24.04/24.10 is an big Change
> 
> Ive only running an Homeassistant OS VM

I can help testing.

**Comment by dmshimself** [comment]
Me too. I was used to doing a new machine and copying of data from previous
issues so if I had to do that again, no probl

**Comment by MickLesk** [comment]
I need some feedback from everyone who feels confident with this.

**BEFORE**: Please make sure to back up your container or clone it.

The goal here is to upgrade from Ubuntu 24.04 to Ubuntu 24.10.

https://github.com/community-scripts/ProxmoxVE/discussions/1549

**Reply by trenchcoatjedi** [reply]
I followed your guide and successfully upgraded to Ubuntu 24.10 but the script still fails to upgrade home assistant in the same way as in the OP.
`[ERROR] in line 64: exit code 0: while executing command uv pip install ${BR}--upgrade homeassistant &> /dev/null`

**Reply by MickLesk** [reply]
where you see an uv install code? you should only execute the guide, not the shell (update or bash call). The Shell-Script **IS NOT MERGED**!

**Reply by trenchcoatjedi** [reply]
The upgrade part of the guide works fine. Am not able to upgrade HA though.

**Reply by MickLesk** [reply]
the Shellscript will be merged if i get 2 another okay's 😄
